client/src/app/pages/authorization/sign-up/sign-up.component.ts

```diff
@@ -12,15 +12,21 @@ export class SignUpComponent implements OnInit {
   private _email: string;
   private _password: string;
 
-  @Output() onSignUp = new EventEmitter<User>();
+  @Output() onSignUp = new EventEmitter<Object>();
+  @Output() onChangeMode = new EventEmitter<User>();
 
   constructor() { }
 
   ngOnInit() {
   }
 
   handleClick():void {
-    this.onSignUp.emit({nickName: this._nickName, email: this._email, password: this._password});
+    this.onSignUp.emit({userName: this._nickName, email: this._email, password: this._password});
+  }
+
+  changeSignMode(modeType): void {
+    document.getElementById('menu').classList.add("anim2");
```

Comment:
Direct DOM manipulation is bad idea. Here probably you can use angular's animation.